webextensions/api/proxy.json

```diff
@@ -114,7 +114,9 @@
               },
               "firefox": [
                 {
-                  "version_added": "56"
+                  "version_added": "56",
+                  "version_removed": "71",
+                  "notes": "Deprecated in Firefox 68 and will be removed entirely in Firefox 71."
```

Comment:
I understand the motivation here (to give developers a clear heads-up that this is coming) but it feels weird making statements about the future in BCD. 71 isn't even Nightly yet.

@Elchi3 , @ddbeck , is there any policy for this?

Assuming it's generally OK to use BCD in this way, I think it feels weird to say "removed entirely in Firefox 71" when the data already says that. I would want the note to say just "Deprecated in Firefox 68".

Reply:
We don't yet have [a policy](https://github.com/mdn/browser-compat-data/blob/master/docs/data-guidelines.md#data-guidelines) for this. But we _could_, now that we're writing such things down!

Absent a policy, my recommendation is that future version data ought to be backed by some authoritative commitment to ship. In this case, Firefox itself warns that it's being removed in 71, so we've definitely got the authoritative part. At the risk of tautology, a version control commit is one way to show commitment, but not the only way. Is there a history of follow through on deprecation notices like this? If so, let's roll with it. And Will is right about the note, in that case.

(The policy question stands though; I'm interested in @Elchi3's take.)

Reply:
Well, deprecation is actually represented by `status { deprecated: true }` already. However, WebExtension BCD hasn't really made use of the `status` object (this was due to the sad story of standardizing WebExtension APIs).

I would love to see `status` happen for webext data one day. For two main reasons:

- In the rendering on the MDN pages, a deprecation symbol would be rendered. 
- BCD consumers could filter for deprecated webext features. webhint does this for web features and it would be great if we would provide `status` objects for WebExtension data and then have the addons-linter (which is already a BCD customer) read the `deprecated` status and warn add-on authors about it.

```json
"unregister": {
  "__compat": {
    "mdn_url": "https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/proxy/unregister",
    "support": {
      "chrome": {
        "version_added": false
      },
      "edge": {
        "version_added": false
      },
      "firefox": {
        "version_added": "56",
        "version_removed": "71"
      },
      "firefox_android": {
        "version_added": "56"
      },
      "opera": {
        "version_added": false
      }
    },
    "status": {
      "experimental": false,
      "standard_track": true,
      "deprecated": true
    }
  }
}
```

Reply:
> Absent a policy, my recommendation is that future version data ought to be backed by some authoritative commitment to ship. In this case, Firefox itself warns that it's being removed in 71, so we've definitely got the authoritative part,

This has been my practice as well. If we have a strong reasoning that something is going to happen then it is totally okay to enter future versions. If it is very vague, like "might ship in 71", lets not add it until we're actually there.

Reply:
I can understand not saying that a feature will be released in a future version, but if we know that a feature is going away, it makes sense to update the BCD as soon as possible. Of course, it is my PR and I'd like to get it done.

Reply:
I have removed the notes in both cases and added a status section showing that the methods have been deprecated.

Reply:
Will and I spoke yesterday and he mentioned that proxy.onError has been deprecated. Actually, it is onProxyError that has been deprecated. The final change that probably has to be made is that the current proxy.json file lists onProxyError as an alternative name for onError. I think that has to be removed but I am checking to make sure.